src/transformers/models/detr/configuration_detr.py

```diff
@@ -187,6 +188,8 @@ def __init__(
                 backbone_model_type = backbone_config.get("model_type")
                 config_class = CONFIG_MAPPING[backbone_model_type]
                 backbone_config = config_class.from_dict(backbone_config)
+            # set timm attributes to None
+            dilation, backbone, use_pretrained_backbone = None, None, None
```

Comment:
To manage the different configuration options, I'd introduce additional argument checks and have defaults which are None to prevent clashes between timm and transformer backbone expected behaviour. As a user, I'd find it confusing to pass in `use_pretrained_backbone=False` and then have a pretrained backbone loaded. 

i.e. something like this: 

```
    def __init__(
        self,
        use_timm_backbone=True,
        backbone_config=None,
        backbone=None,
        use_pretrained_backbone=None,
        dilation=None, 
    ):
    
    if use_timm_backbone:
        dilation = dilation if dilation is not None else False
        use_pretrained_backbone = use_pretrained_backbone if use_pretrained_backbone is not None else True
        backbone = backbone if backbone is not None else "resnet50"         

        if backbone_config is not None: 
            raise ValueError("Cannot specify backbone_config if use_timm_backbone is None")
        
    else:
        if None in [dilation, use_pretrained_backbone, backbone]:
            raise ValueError(
                "Cannot specify dilation, use_ptrained_backbone or backbone if use_timm_backbone is False"
            )
```

Reply:
Agreed! Note that this can go in a followup PR.